src/v/redpanda/admin/util.cc

```diff
@@ -13,6 +13,27 @@
 
 #include <seastar/http/exception.hh>
 
+namespace {
+
+short hex_to_byte(char c) {
+    if (c >= 'a' && c <= 'z') {
+        return c - 'a' + 10;
+    } else if (c >= 'A' && c <= 'Z') {
+        return c - 'A' + 10;
+    }
+    return c - '0';
+}
+
+/**
+ * Convert a hex encoded 2 bytes substring to char
+ */
+char hexstr_to_char(const std::string_view& in, size_t from) {
```

Comment:
nitpick: Prefer to pass `std::string_view` by value.

Reply:
I left this as is to keep the diff minimal compared to seastar's versions of these methods ([here](https://github.com/scylladb/seastar/blob/ae3fa2dd30e72cb28d5a3705420798d73359a1a3/src/http/url.cc#L51)). I think I'll leave this as is and try to delete these methods once seastar has a `path_decode` method if that sounds good to you too.

Reply:
this PR is add new code so there isn't really a diff against seastar, right? I'd just change it to take a value.

Reply:
Fair enough, I've made the change now: https://github.com/redpanda-data/redpanda/pull/16694/commits/490721980881ae4d11f3934f4ec4417b9351e55f